omnigibson/action_primitives/curobo.py

```diff
@@ -0,0 +1,590 @@
+            lazy.curobo.geom.types.WorldConfig(),  # world_cfg,
+            self._tensor_args,
+            store_trajopt_debug=self.debug,
+            **motion_kwargs,
+        )
+        self.mg = lazy.curobo.wrap.reacher.motion_gen.MotionGen(motion_gen_config)
+        # print("CuRobo warming up...")
+        # self.mg.warmup(enable_graph=False, warmup_js_trajopt=False, batch=batch_size)
+        # print("CuRobo is ready.")
+
+        # Store internal variables
+        self.robot = robot
+        self.ee_link = robot_cfg["kinematics"]["ee_link"]
+        self._fk = FKSolver(self.robot.robot_arm_descriptor_yamls[robot.default_arm], self.robot.urdf_path)
+        self._usd_help = lazy.curobo.util.usd_helper.UsdHelper()
+        self._usd_help.stage = og.sim.stage
+        self._ignore_collision_paths = [] if ignore_collision_paths is None else ignore_collision_paths
+        assert batch_size >= 2, f"batch_size must be >= 2! Got: {batch_size}"
+        self.batch_size = batch_size
+
+        # Initialize other variables modified at runtime
+        self._ignore_paths = None
+
+    def update_obstacles(self):
+        """
+        Updates internal world collision cache representation based on sim state
+        """
+        print("Updating CuRobo world, reading w.r.t.", self.robot.prim_path)
+        # Infer how many collision bodies there are in sim, and add 10 as an additional buffer
+        n_cols = 0
+
+        # Ignore all container metalinks
+        self._ignore_paths = ["container"]
+
+        # Ignore any visual only objects and any objects not part of the robot's current scene
+        for idx, scene in enumerate(og.sim.scenes):
+            if idx == self.robot.scene.idx:
+                for obj in scene.objects:
+                    if obj.visual_only:
+                        self._ignore_paths.append(obj.prim_path)
+                        continue
+                    for link in obj.links.values():
+                        n_cols += len(link.collision_meshes)
+            else:
+                self._ignore_paths.append(scene.prim_path)
+
+        # # TODO: This should be modified dynamically but it breaks )):
+        # assert isinstance(self.mg.world_coll_checker, WorldMeshCollision)
+        # self.mg.world_coll_checker.clear_cache()
+        # self.mg.world_coll_checker.create_collision_cache(mesh_cache=n_cols + 10)
+
+        obstacles = self._usd_help.get_obstacles_from_stage(
+            reference_prim_path=self.robot.root_link.prim_path,
+            ignore_substring=[
+                self.robot.prim_path,  # Don't include robot paths
+                "/World/ground_plane",  # Don't include collisions with floor
+                "visual",  # Don't include any visuals
+                "/curobo",  # Don't include curobo prim
+                *self._ignore_collision_paths,  # Don't include manually-specified collision paths
+                *self._ignore_paths,  # Don't include any additional specified paths
```

Comment:
is this actually needed? I felt like there are some "duplicated" exclusion logic here.